vcs/gitcmd/repo.go

```diff
@@ -401,6 +409,12 @@ func (r *Repository) commitLog(opt vcs.CommitsOptions) ([]*vcs.Commit, uint, err
 
 	// Count commits.
 	cmd = exec.Command("git", "rev-list", "--count", rng)
+	if opt.Path != "" {
+		// TODO: This doesn't include --follow flag because rev-list doesn't support it, so the number may be slightly off.
+		//       Consider removing the "total" return value from the API altogether, that's why I'm not trying to do more to produce an accurate number here.
+		//       Getting the total number of commits is an expensive operation and likely isn't needed, or deserves a separate dedicated API if needed.
+		cmd = exec.Command("git", "rev-list", "--count", rng, "--", opt.Path)
```

Comment:
Ah, nice, you're familiar with `exec.Cmd` internals too. :P I didn't do this because I thought it might be less readable. But it doesn't matter because this code will be deleted soon once we remove total commit count from Commits API.